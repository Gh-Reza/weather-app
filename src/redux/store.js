import { configureStore, applyMiddleware } from '@reduxjs/toolkit';
import countriesReducer from './countries/countriesSlice';

const store = configureStore({
  reducer: {
    countries: countriesReducer,
  },
}, applyMiddleware);

export default store;
