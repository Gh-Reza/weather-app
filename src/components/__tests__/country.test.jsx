import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../../redux/store';
import '@testing-library/jest-dom/extend-expect';
import Country from '../country';

describe('Tests the country component', () => {
  const component = (
    <BrowserRouter>
      <Provider store={store}>
        <Country
          key="Afghnistan"
          name="Afghnistan"
          flag=""
          capital="Kabul"
          lat={12}
          lng={3}
          onClick={() => {}}
        />
      </Provider>
    </BrowserRouter>
  );

  test('Match the snapshot', () => {
    const tree = render(component);
    expect(tree).toMatchSnapshot();
  });
});
