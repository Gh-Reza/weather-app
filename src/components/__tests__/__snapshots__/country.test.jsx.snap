// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests the country component Match the snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="col p-0 country-box"
      >
        <a
          class="text-decoration-none text-light"
          href="/details/Afghnistan"
        >
          <div
            class="country container p-4"
          >
            <div
              class="row"
            >
              <div
                class="col-12"
              >
                <h3
                  class="country-name h1"
                >
                  Afghnistan
                </h3>
                <p
                  class="capital-name"
                >
                  Kabul
                </p>
              </div>
              <div
                class="col-12 d-flex flag-box text-end"
              >
                <img
                  alt=""
                  class="flag img-thumbnail border border-danger"
                  src=""
                />
                <div
                  class="d-flex flex-column align-items-start justify-content-center ms-3"
                >
                  <p
                    class="h5"
                  >
                    12
                  </p>
                  <p
                    class="h5"
                  >
                    3
                  </p>
                </div>
              </div>
            </div>
          </div>
        </a>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="col p-0 country-box"
    >
      <a
        class="text-decoration-none text-light"
        href="/details/Afghnistan"
      >
        <div
          class="country container p-4"
        >
          <div
            class="row"
          >
            <div
              class="col-12"
            >
              <h3
                class="country-name h1"
              >
                Afghnistan
              </h3>
              <p
                class="capital-name"
              >
                Kabul
              </p>
            </div>
            <div
              class="col-12 d-flex flag-box text-end"
            >
              <img
                alt=""
                class="flag img-thumbnail border border-danger"
                src=""
              />
              <div
                class="d-flex flex-column align-items-start justify-content-center ms-3"
              >
                <p
                  class="h5"
                >
                  12
                </p>
                <p
                  class="h5"
                >
                  3
                </p>
              </div>
            </div>
          </div>
        </div>
      </a>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
